modules/gapi/samples/semantic_segmentation.cpp

```diff
@@ -148,29 +146,46 @@ int main(int argc, char *argv[]) {
     // Now build the graph
     cv::GMat in;
     cv::GMat out_blob = cv::gapi::infer<SemSegmNet>(in);
-    cv::GMat out = custom::PostProcessing::on(in, out_blob);
+    cv::GMat post_proc_out = custom::PostProcessing::on(in, out_blob);
+    cv::GMat blending_in = cv::gapi::mulC(in, 0.3, CV_8U);
+    cv::GMat blending_out = cv::gapi::mulC(post_proc_out, 0.7, CV_8U);
+    cv::GMat out = cv::gapi::add(blending_in, blending_out, CV_8U);
 
     cv::GStreamingCompiled pipeline = cv::GComputation(cv::GIn(in), cv::GOut(out))
         .compileStreaming(cv::compile_args(kernels, networks));
     auto inputs = cv::gin(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(input));
 
     // The execution part
     pipeline.setSource(std::move(inputs));
-    pipeline.start();
 
     cv::VideoWriter writer;
+    cv::VideoCapture cap(input);
+
+    if (!output.empty()) {
+        cv::Mat first_frame;
+        cap >> first_frame;
+        const auto sz = cv::Size{ first_frame.cols, first_frame.rows };
+        cap.release();
+        writer.open(output, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'), 25.0, sz);
+    }
```

Comment:
Why do we need this here? Only to initialize Writer outside of the measurements?

Reply:
I believe the old approach also worked isnt it?

Reply:
Returned previous variant.